walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,118 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Iterator;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
```

Comment:
This is a really long function. Try to break it into smaller, logical chunks and separate them out into functions to improve readability.

Reply:
Okay, sounds good!

Reply:
As of now, there are certain conditions that are specific to the method. Do you think it would still be worth to break it into functions if it's just using the same code in different functions?

Reply:
Yeah I think even if it doesn't make the class any shorter, using helper methods can make it easier to read. For example, where you have the comment `// Gets rid of any possible events in the array or impossible time ranges.`, you could change it to a helper function, something like:

```
/** Gets rid of any possible events in the array or impossible time ranges. */
private void removeImpossibleTimes(ArrayList<TimeRange> times) {...}
```